services/QuillLMS/engines/evidence/spec/controllers/evidence/activated_llm_prompts_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Evidence
+  RSpec.describe ActivatedLLMPromptsController, type: :controller do
+    routes { Evidence::Engine.routes } # Use the correct engine routes
+
+    let(:json_response) { JSON.parse(response.body) }
+    let(:prompt) { create(:evidence_prompt) }
+    let(:llm_prompt) { create(:evidence_research_gen_ai_llm_prompt) }
+    let(:valid_params) do
+      {
+        research_gen_ai_activated_llm_prompt: {
+          active: true,
+          llm_prompt_id: llm_prompt.id,
+          notes: 'Sample note'
+        },
+        prompt_id: prompt.id
+      }
+    end
+
+    describe 'GET #index' do
+      let!(:activated_llm_prompt) { create(:evidence_activated_llm_prompt, llm_prompt:, prompt:, active: true) }
+
+      it 'returns a successful response' do
+        get :index, params: { prompt_id: prompt.id }
+
+        expect(response).to be_successful
+        expect(json_response['activated_llm_prompts']).to be_present
+        expect(json_response['activated_llm_prompts'].first['id']).to eq(activated_llm_prompt.id)
+      end
+
+      it 'includes associated trial data' do
+        trial = create(:evidence_research_gen_ai_trial, llm_prompt:)
+        activated_llm_prompt.update!(llm_prompt:)
+
+        get :index, params: { prompt_id: prompt.id }
+
+        expect(response).to be_successful
+        expect(json_response['activated_llm_prompts'].first['trial']).to be_present
+        expect(json_response['activated_llm_prompts'].first['trial']['id']).to eq(trial.id)
+      end
+    end
+
+    describe 'POST #create' do
+      context 'with valid parameters' do
+        it 'creates a new activated LLM prompt' do
+          expect {
+            post :create, params: valid_params
+          }.to change(Evidence::ActivatedLLMPrompt, :count).by(1)
+
+          expect(response).to have_http_status(:ok)
+          expect(json_response['id']).to be_present
+        end
+
+        it 'sets other prompts to inactive if active is true' do
+          existing_prompt = create(:evidence_activated_llm_prompt, llm_prompt:, prompt:, active: true)
```

Comment:
Again, take or leave style thing, but I like `create` calls in `let` which means you'd need a `context` wrapper.